src/modules/map/components/cesium/CesiumGPXLayer.vue

```diff
@@ -0,0 +1,159 @@
+<script setup>
+import {
+    BillboardGraphics,
+    Cartesian3,
+    Color,
+    ColorMaterialProperty,
+    defined as cesiumDefined,
+    GpxDataSource,
+    HeightReference,
+} from 'cesium'
+import { onMounted, onUnmounted, ref, watch } from 'vue'
+import { inject } from 'vue'
+
+import GPXLayer from '@/api/layers/GPXLayer.class'
+import log from '@/utils/logging'
+
+const props = defineProps({
+    gpxLayerConfig: {
+        type: GPXLayer,
+        required: true,
+    },
+})
+
+const isPresentOnMap = ref(false)
+const gpxDataSource = ref(new GpxDataSource())
+
+const opacity = ref(props.gpxLayerConfig.opacity)
+const gpxData = ref(props.gpxLayerConfig.gpxData)
+
+const getViewer = inject('getViewer')
+watch(gpxData, () => {
+    addLayer()
+})
+
+watch(
+    () => props.gpxLayerConfig.opacity,
+    (newOpacity) => {
+        opacity.value = newOpacity
+        updateStyle()
+    }
+)
+
+onMounted(() => {
+    log.debug('Mounted GPX layer')
+    addLayer()
+})
+
+onUnmounted(() => {
+    log.debug('Unmounted GPX layer')
+    if (gpxDataSource.value && isPresentOnMap.value) {
+        removeLayer()
+    }
+
+    gpxDataSource.value = null
+})
+
+function addLayer() {
+    const gpxBlob = new Blob([gpxData.value], { type: 'application/gpx+xml' })
+    const gpxUrl = URL.createObjectURL(gpxBlob)
+
+    gpxDataSource.value
+        .load(gpxUrl, {
+            clampToGround: true,
+        })
+        .then((dataSource) => {
+            getViewer().dataSources.add(dataSource)
+            isPresentOnMap.value = true
+        })
+        .then(updateStyle)
+}
+
+function removeLayer() {
+    log.debug('Remove GPX layer')
+    if (gpxDataSource.value) {
+        getViewer().dataSources.remove(gpxDataSource.value)
+        gpxDataSource.value = null
+        getViewer().scene.requestRender() // Request a render after removing the DataSource
+    }
+    isPresentOnMap.value = false
+}
+
+function updateStyle() {
+    // Function to create a red circle image using a canvas
+    function createRedCircleImage(radius) {
+        // Create a new canvas element
+        const canvas = document.createElement('canvas')
+        const context = canvas.getContext('2d')
+
+        // Set the canvas sizes
+        canvas.width = radius * 2
+        canvas.height = radius * 2
+
+        // Draw a red circle on the canvas
+        context.beginPath()
+        context.arc(radius, radius, radius, 0, 2 * Math.PI, false)
+
+        const color = `rgba(255, 0, 0, ${opacity.value})`
+        context.fillStyle = color
+        context.fill()
+
+        // Return the data URL of the canvas drawing
+        return canvas.toDataURL()
+    }
```

Comment:
you might want to have a look at this https://github.com/geoadmin/web-mapviewer/blob/37ce862f42c3359b80ee50bb00e50066bcc864df/src/modules/map/components/cesium/CesiumGeolocationFeedback.vue#L84-L111

where a circle entity is created using only Cesium stuff (no canvas).

Reply:
Cesium load the point as a billboard. Should I change it to circle/ellipse in this case?

Reply:
We will use this approach per my discussion with @pakb